core/src/test/java/org/apache/iceberg/TestUpdateSnapshotRefs.java

```diff
@@ -0,0 +1,255 @@
+    table.updateRefs()
+        .setBranchSnapshotLifetime(TEST_BRANCH, 10000)
+        .commit();
+    table.refresh();
+    Assert.assertEquals("Should have latest snapshot lifetime config for branch",
+        10000, (long) table.refs().get(TEST_BRANCH).maxSnapshotAgeMs());
+  }
+
+  @Test
+  public void testSetMinSnapshotsInBranch() {
+    table.updateRefs()
+        .setMinSnapshotsInBranch(TEST_BRANCH, 10000)
+        .commit();
+    table.refresh();
+    Assert.assertEquals("Should have latest snapshot in branch config for branch",
+        10000, (int) table.refs().get(TEST_BRANCH).minSnapshotsToKeep());
+  }
+
+  @Test
+  public void testRenameTag() {
+    SnapshotRef tagRef = table.refs().get(TEST_TAG);
+    table.updateRefs().rename(TEST_TAG, "tag2").commit();
+    Assert.assertEquals("Renamed tag should have the same config", tagRef, table.refs().get("tag2"));
+  }
+
+  @Test
+  public void testRenameBranch() {
+    SnapshotRef tagRef = table.refs().get(TEST_BRANCH);
+    table.updateRefs().rename(TEST_BRANCH, "branch2").commit();
+    Assert.assertEquals("Renamed branch should have the same config", tagRef, table.refs().get("branch2"));
+  }
+
+  @Test
+  public void testInvalidRename() {
+    AssertHelpers.assertThrows("Should not have null from name",
+        IllegalArgumentException.class,
+        "Names must not be null",
+        () -> table.updateRefs().rename(null, "to"));
+
+    AssertHelpers.assertThrows("Should not have null to name",
+        IllegalArgumentException.class,
+        "Names must not be null",
+        () -> table.updateRefs().rename("from", null));
+
+    AssertHelpers.assertThrows("From ref must exist",
+        IllegalArgumentException.class,
+        "Cannot find ref to rename from: tag",
+        () -> table.updateRefs().rename("tag", "tag2"));
+
+    AssertHelpers.assertThrows("To ref must not exist",
+        IllegalArgumentException.class,
+        "Cannot rename to an existing ref: " + TEST_BRANCH,
+        () -> table.updateRefs().rename(TEST_TAG, TEST_BRANCH));
+  }
+
+  @Test
+  public void testRemoveTag() {
+    table.updateRefs().remove(TEST_TAG).commit();
+    Assert.assertNull("Removed tag should not exist", table.refs().get(TEST_TAG));
+  }
+
+  @Test
+  public void testRemoveBranch() {
+    table.updateRefs().remove(TEST_BRANCH).commit();
+    Assert.assertNull("Removed branch should not exist", table.refs().get(TEST_BRANCH));
+  }
+
+  @Test
+  public void testRemoveMainBranchShouldFail() {
+    AssertHelpers.assertThrows("Should fail when removing main branch",
+        IllegalArgumentException.class,
+        "Main branch must not be removed",
+        () -> table.updateRefs().remove(SnapshotRef.MAIN_BRANCH));
+  }
+
+  @Test
+  public void testRemoveRefNotExist() {
+    AssertHelpers.assertThrows("Should fail when removing ref not exist",
+        IllegalArgumentException.class,
+        "Cannot find ref to remove",
+        () -> table.updateRefs().remove("tag"));
+  }
```

Comment:
I think we are missing a test for chaining multiple operations for tag/branch, like

```
table.updateRefs()
  .branch("b", 123)
  .setMinSnapshotsInBranch("b", 10000)
  .build()
```

to make sure chaining works as expected when creating new ref and set values at the same time.